lib/puppet/provider/cisco_bgp/nxapi.rb

```diff
@@ -200,11 +199,6 @@ def timer_bestpath_limit_set
     @bgp_vrf.timer_bestpath_limit_set(limit, always)
   end
 
-  def confederation_peers_set
-    return unless @property_flush[:confederation_peers]
-    @bgp_vrf.confederation_peers_set(@property_flush[:confederation_peers])
-  end
-
   def flush
     if @property_flush[:ensure] == :absent
       @bgp_vrf.destroy
```

Comment:
I think that is part of the special handling for nested arrays as well:

i.e:

 # networks/redistribute use nested arrays, thus require special handling
    current_state[:networks] = obj.networks
    current_state[:redistribute] = obj.redistribute